autoextract/aio/client.py

```diff
@@ -9,24 +9,26 @@
 from functools import partial
 
 import aiohttp
+from aiohttp import TCPConnector
 
 from autoextract.constants import API_ENDPOINT, API_TIMEOUT
 from autoextract.apikey import get_apikey
 from autoextract.utils import chunks, user_agent
 from autoextract.request import Query, query_as_dict_list
 from autoextract.stats import ResponseStats, AggStats
 from .retry import autoextract_retry, QueryRetryError
-from .errors import RequestError, _QueryError
-
+from .errors import RequestError, _QueryError, is_billable_error_msg
 
 AIO_API_TIMEOUT = aiohttp.ClientTimeout(total=API_TIMEOUT + 60,
                                         sock_read=API_TIMEOUT + 30,
                                         sock_connect=10)
 
 
-def create_session(**kwargs) -> aiohttp.ClientSession:
+def create_session(connection_pool_size=100, **kwargs) -> aiohttp.ClientSession:
     """ Create a session with parameters suited for AutoExtract """
     kwargs.setdefault('timeout', AIO_API_TIMEOUT)
+    if "connector" not in kwargs:
+        kwargs["connector"] = TCPConnector(limit=connection_pool_size)
```

Comment:
Why is this needed? I was hoping that we're using http 1.1 and sending all requests in a single tcp connection; we're connectingto a single host.

Reply:
@kmike This is a good point. I can say that I have tested to set TCPConnector(limit=1) and performance is dramatically affected. It seems requests are sent in sequence in such a case. Are you sure that under Http1.1 it is possible to mix request/responses under the same connection in a parallel way? I mean, requests are never blocked, and responses are being read as they come (some kind of streaming). From [Wikipedia](https://en.wikipedia.org/wiki/Chunked_transfer_encoding):

> Chunked encoding has the benefit that it is not necessary to generate the full content before writing the header, as it allows streaming of content as chunks and explicitly signaling the end of the content, **making the connection available for the next HTTP request/response**.

According to that still under Http1.1 requests are executed in sequence it seems. So this is why the pool size matters.

Reply:
I think you're right @ivanprado, and that's a correct change.